pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -229,6 +229,17 @@ private void manageSegmentLineageCleanupForTable(String tableNameWithType) {
               // If the lineage state is 'COMPLETED', it is safe to delete all segments from 'segmentsFrom'
               segmentsToDelete.addAll(sourceSegments);
             }
+          } else if (lineageEntry.getState() == LineageEntryState.REVERTED) {
```

Comment:
We can simplify it because `REVERTED` should have the same behavior as expired `IN_PROGRESS`
```suggestion
          } else if (lineageEntry.getState() == LineageEntryState.REVERTED || (lineageEntry.getState() == LineageEntryState.IN_PROGRESS && lineageEntry.getTimestamp() < System.currentTimeMillis() - LINEAGE_ENTRY_CLEANUP_RETENTION_IN_MILLIS)) {
```

Reply:
good catch. I merged the `IN_PROGRESS` and `REVERTED`